File: RemoteMonitor Desktop/src/main/java/components/CPU.java
package components;
import oshi.SystemInfo.*;
import oshi.hardware.CentralProcessor;
import oshi.hardware.Sensors;

public class CPU extends Component {
    private int cores;
    private long speed;
    private long[] currentSpeed;
    private int[] fanSpeeds;
    private transient CentralProcessor processor;
    private transient Sensors sensors;
    public CPU()
    {
        processor = si.getHardware().getProcessor();
        cores = processor.getPhysicalProcessorCount();
        componentName = processor.getProcessorIdentifier().getName();
        componentType = "CPU";
        speed = processor.getMaxFreq();
        sensors = si.getHardware().getSensors();
    }
    public void update() {
        usage = processor.getSystemCpuLoad(1000);
        temperature = sensors.getCpuTemperature();
        currentSpeed = processor.getCurrentFreq();
        fanSpeeds = sensors.getFanSpeeds();
        speed = processor.getMaxFreq();
        lastUpdated = System.currentTimeMillis();
    }
    public long getSpeed()
    {
        return speed;
    }
    public long[] getCurrentSpeed()
    {
        return currentSpeed;
    }
    public int getCores() {
        return cores;
    }
}

File: RemoteMonitor Mobile/RemoteMonitor/RemoteMonitor/RemoteMonitor/Components/CPU.cs
using Newtonsoft.Json.Linq;
using System;
using System.Collections.Generic;
using System.Diagnostics;
using System.Text;
using static Xamarin.Essentials.Permissions;

namespace RemoteMonitor.Components
{
    public class CPU : Component
    { 
        public int cores;
        public long speed;
        public long[] currentSpeed;
        public int[] fanSpeed;
        public CPU(string name, double temp, double componentUsage, int coreAmount, long maxSpeed, long[] currentSpeeds, int[] fanSpeeds)
        {
            componentName = name;
            componentType = "CPU";
            update(temp, componentUsage, coreAmount, maxSpeed, currentSpeeds, fanSpeeds);
        }
        public CPU(JObject json) 
        {
            componentName = json["componentName"].ToString();
            componentType = "CPU";

            update(json);
         
        }
        public void update(double temp, double componentUsage, int coreAmount, long maxSpeed, long[] currentSpeeds, int[] fanSpeeds)
        {
            temperature= temp;
            usage = componentUsage;
            cores = coreAmount;
            speed = maxSpeed;
            currentSpeed= currentSpeeds;
            fanSpeed = fanSpeeds;

        }
        public void update(JObject json) 
        {
            temperature = ((double)json["temperature"]);
            usage = ((double)json["usage"]);
            cores = json["cores"].ToObject<int>();
            speed = json["speed"].ToObject<long>();
            currentSpeed = json["currentSpeed"].ToObject<long[]>();
            fanSpeed = json["fanSpeeds"].ToObject<int[]>();
        }
        public int[] getFanSpeeds()
        {
            return fanSpeed;
        }
        public long getSpeed()
        {
            return speed;
        }
        public long[] getCurrentSpeed()
        {
            return currentSpeed;
        }
        public int getCores()
        {
            return cores;
        }
    }
}
